### PR TITLE
[Paddle-TRT] IPluginExt -> IPluginV2

### PR DESCRIPTION
**Pull request by zlsh80826**
### PR types

Breaking changes

### PR changes

Others

### Describe

IPluginExt is deprecated. This PR will upgrade IPluginExt to IPluginV2

### CODE REVIEW

File: paddle/fluid/inference/tensorrt/plugin/instance_norm_op_plugin.cu [hunk 1]
```diff
@@ -58,6 +51,13 @@ nvinfer1::Dims InstanceNormPlugin::getOutputDimensions(
   return output_dims;
 }
 
+bool InstanceNormPlugin::supportsFormat(nvinfer1::DataType type,
+                                        nvinfer1::PluginFormat format) const {
+  return ((type == nvinfer1::DataType::kFLOAT ||
+           type == nvinfer1::DataType::kHALF) &&
+          (format == nvinfer1::PluginFormat::kNCHW));
```
Comment (shangzhizhou) [inline paddle/fluid/inference/tensorrt/plugin/instance_norm_op_plugin.cu]: nvinfer1::PluginFormat::kLINEAR?

File: paddle/fluid/inference/tensorrt/plugin/trt_plugin_factory.cc [hunk 2]
```diff
@@ -21,6 +21,7 @@ namespace inference {
 namespace tensorrt {
 namespace plugin {
 
+#if false
```
Comment (shangzhizhou) [inline paddle/fluid/inference/tensorrt/plugin/trt_plugin_factory.cc]: 没用的代码可以删掉

File: paddle/fluid/inference/tensorrt/plugin/trt_plugin_factory.h [hunk 3]
```diff
@@ -33,6 +33,7 @@ namespace inference {
 namespace tensorrt {
 namespace plugin {
 
+#if false
```
Comment (shangzhizhou) [inline paddle/fluid/inference/tensorrt/plugin/trt_plugin_factory.h]: 没用的代码可以删掉

File: paddle/fluid/inference/tensorrt/plugin/prelu_op_plugin.h [hunk 4]
```diff
@@ -101,7 +113,6 @@ class PReluPluginDynamic : public DynamicPluginTensorRT {
   // It was used for tensorrt deserialization.
   // It should not be called by users.
   PReluPluginDynamic(void const* serialData, size_t serialLength) {
```
Comment (shangzhizhou) [inline paddle/fluid/inference/tensorrt/plugin/prelu_op_plugin.h]: PReluPluginDynamic的序列化返回空字符串，辛苦排查修复该PR涉及的几个plugin的类似问题

File: paddle/fluid/inference/tensorrt/plugin/trt_plugin_factory.h (hunk 3 above)
Comment (zlsh80826) [reply paddle/fluid/inference/tensorrt/plugin/trt_plugin_factory.h]: done

File: paddle/fluid/inference/tensorrt/plugin/prelu_op_plugin.h (hunk 4 above)
Comment (zlsh80826) [reply paddle/fluid/inference/tensorrt/plugin/prelu_op_plugin.h]: PRelu & Pool dynamic plugin 已修復並加上 ut

**Review by shangzhizhou — APPROVED** [review]
LGTM

**Review by raindrops2sea — APPROVED** [review]
LGTM

**Review by cryoco — APPROVED** [review]
LGTM

**Pull request merged**